src/cljs/athens/views/blocks.cljs

```diff
@@ -517,10 +560,19 @@
                                                    (js/setTimeout #(swap! state assoc :string/local new-str) 50)))
                   (re-find #"text/html" datatype) (.getAsString item (fn [_] #_(prn "getAsString" _))))))
             items)
-      :else
-      (when (and line-breaks no-shift)
+
+      (and line-breaks no-shift)
+      (do
         (.. e preventDefault)
-        (dispatch [:paste uid text-data])))))
+        (dispatch [:paste uid text-data]))
+
+      (not no-shift)
+      (do
+        (.. e preventDefault)
+        (dispatch [:paste-verbatim uid text-data]))
```

Comment:
To fix: don't overwrite existing content. Make sure inserts to not simply append either, in case paste-verbatim happens not at the end of the existing text content

Reply:
Good catch, forgot about pasting in the middle of existing text.

Reply:
Verbatim Paste now respects selection.